test/ForwardMode/NotEnoughArgError.C

```diff
@@ -0,0 +1,14 @@
+// RUN: %cladclang %s -I%S/../../include -fsyntax-only -Xclang -verify 2>&1
+// CHECK-NOT: {{.*warning|note:.*}}
+// XFAIL:*
+
+#include "clad/Differentiator/Differentiator.h"
+
+double func1(double x, double y) {
+  return x * x + y * y;
+}
+
+int main () {
+  auto func1_dx = clad::differentiate(func1, "x");
+  func1_dx.execute(5);
```

Comment:
```suggestion
  func1_dx.execute(5); // expected-error {{some text here}}
```

Please remove the XFAIL.

Reply:
The error is not thrown there:
```cpp
/media/sudo-panda/D/workspaces/clad/src/tools/clad/include/clad/Differentiator/Differentiator.h:113:40: error: too few arguments to function call, expected 2, have 1
    return f(static_cast<Args>(args)...);
           ~                           ^
/media/sudo-panda/D/workspaces/clad/src/tools/clad/include/clad/Differentiator/Differentiator.h:245:16: note: in instantiation of function template specialization 'clad::execute_with_default_args<false, double, double (*)(double, double), int, true>' requested here
        return execute_with_default_args<EnablePadding>(
               ^
/media/sudo-panda/D/workspaces/clad/src/tools/clad/include/clad/Differentiator/Differentiator.h:196:14: note: in instantiation of function template specialization 'clad::CladFunction<double (*)(double, double), clad::NoObject, false>::execute_helper<double (*)(double, double), int>' requested here
      return execute_helper(m_Function, static_cast<Args>(args)...);
             ^
../src/tools/clad/test/ForwardMode/Errors.C:13:12: note: in instantiation of function template specialization 'clad::CladFunction<double (*)(double, double), clad::NoObject, false>::execute<int, double (*)(double, double)>' requested here
  func1_dx.execute(5); //expected-error {{too few arguments to function call, expected 2, have 1}}
           ^
```

Reply:
Works!